File: 0848-shifting-letters/0848-shifting-letters.java
class Solution {
    public String shiftingLetters(String s, int[] shifts) {
        
        char[] chars = s.toCharArray();
        long sum = 0;
        for(int i = chars.length-1; i >=0; i--){
            sum += shifts[i];
            long temp = (chars[i] - 'a' + sum)%26;
            chars[i] = (char) ((int)temp + 'a');
        }    
        
        return String.valueOf(chars);
    }
}

File: 0848-shifting-letters/0848-shifting-letters.kt
class Solution {
    fun shiftingLetters(s: String, shifts: IntArray): String {
        val chars = s.toCharArray()
        var sum : Long = 0
        for(i in chars.size-1 downTo 0){
            sum += shifts[i]
            val temp = (chars[i]-'a' + sum)%26
            chars[i] = (temp + 'a'.toInt()).toChar()
        }
        return String(chars)
    }
}